### ISSUE TITLE
Discard master/slave jargon in pysmurf (where possible)

### ISSUE THREAD

**Issue by swh76** [slaclab/pysmurf#471]
After discussing this outside PR:

https://github.com/slaclab/pysmurf/pull/470

we've agreed to eliminate the use of master/slave jargon where possible in pysmurf (in some places it will not be possible, at least in the short term, because pysmurf is largely a rogue API,  but folks are thinking about how to eliminate the use of the terms there as well).

**Comment by swh76** [comment]
We also need to rename the `master` branch.

**Comment by swh76** [comment]
I just renamed the `master` branch to `main`.

**Comment by ruck314** [comment]
PR #931 addresses what pysmurf can fix without breaking user scripts. Recapping the scope so we can decide whether this issue should stay open:

### Fixed in #931

- `docs/conf.py`: `master_doc` → `root_doc`.
- 3 stale `/master/` URLs flipped to `/main/`: our own `README.SmurfPacket.md`, `simonsobs/ocs`, and `actions/labeler`.
- 9 pyrogue `Variable` descriptions: `"…pass thorough to the next slave."` → `"…pass through to the next stage."` (also fixes the typo).
- Internal class renames (all private, never imported externally): `DataMaster`/`DataSlave`/`MetaSlave` → `DataSource`/`DataSink`/`MetaSink`; `_data_master`/`_data_slave`/`_meta_slave`/`_stream_slaves` → `_data_source`/`_data_sink`/`_meta_sink`/`_stream_sinks`.
- Public API rename with **deprecation aliases** so user scripts don't break:
  - `load_master_assignment` → `load_channel_assignment`
  - `get_master_assignment` → `get_channel_assignment`
  - `write_master_assignment` → `write_channel_assignment`
  - `make_master_assignment_from_file` → `make_channel_assignment_from_file`
  - `new_master_assignment=` kwarg on `tune`/`tune_band_serial`/`assign_channels`/`setup_notches`/`noise_vs_tone` → `new_channel_assignment=`

  The old names still work and emit a `DeprecationWarning`.

### Still using `master`/`slave` after #931, with reason

- All `class X(rogue.interfaces.stream.Master/Slave)` inheritance and `_getStreamMaster`/`_getStreamSlave` method names — that's the Rogue API contract. As of v6.11.0 (verified at https://slaclab.github.io/rogue/latest/stream_interface/index.html) Rogue still ships these names.
- `pub.register_file(fn, 'master_assignment', …)` — that string is consumed by `pysmurf-archiver`. Renaming silently drops files from existing archives, so I left it.
- `cfg_files/**/*.cfg` and `smurf_config.py:770` `"timing master"` references — that's the SLAC ATCA AMC timing card, hardware terminology shared with surf / AmcCarrierCore / firmware. Coordinating that rename is out of scope here.
- `python/pysmurf/client/util/pub.py:6` `mhasself/stdpub` URL — upstream is still on `master` (their `main` returns 404).
- `scratch/cyu/tune_scratch.py` — frozen scratch fork; calls into pysmurf still work via the deprecated aliases.

### Recommendation

Keep this issue open. The remaining work is gated by upstream Rogue's API rename — close once Rogue itself moves and pysmurf can drop the `Master`/`Slave` inheritance / `_getStream*` references along with our deprecation aliases.
